src/main/scala/com/atamaniv/RetryImpl.scala

```diff
@@ -0,0 +1,28 @@
+package com.atamaniv
+
+import scala.annotation.tailrec
+import scala.concurrent.duration.FiniteDuration
+
+class RetryImpl extends Retry {
+
+  @tailrec
+  override final def retry[A](block: () => A, acceptResult: A => Boolean, retries: List[FiniteDuration]): A = {
+    val result = block()
+
+    if (acceptResult(result))
+      result
+    else retries match {
+      case head :: Nil =>
+        sleepFor(head.toMillis)
+        retry(block, acceptResult, Nil)
+      case head :: tail =>
+        sleepFor(head.toMillis)
+        retry(block, acceptResult, tail)
```

Comment:
what is the difference between 2 cases above? it looks like the first case is redundant